gdk/commands/component/InitCommand.py

```diff
@@ -0,0 +1,135 @@
+            repository = self.command_args["repository"]
+            if repository:
+                logging.info(
+                    "Initializing the project directory with a component from repository catalog - '{}'.".format(repository)
+                )
+                self.init_with_repository(repository, project_dir)
+                return
+        raise Exception(error_messages.INIT_WITH_INVALID_ARGS)
+
+    def init_with_template(self, template, language, project_dir):
+        try:
+            template_name = "{}-{}".format(template, language)
+            logging.info("Fetching the component template '{}' from Greengrass Software Catalog.".format(template_name))
+            self.download_and_clean(template_name, "template", project_dir)
+        except Exception as e:
+            raise Exception("Could not initialize the project with component template '{}'.\n{}".format(template, e))
+
+    def init_with_repository(self, repository, project_dir):
+        try:
+            logging.info("Fetching the component repository '{}' from Greengrass Software Catalog.".format(repository))
+            self.download_and_clean(repository, "repository", project_dir)
+        except Exception as e:
+            raise Exception("Could not initialize the project with component repository '{}'.\n{}".format(repository, e))
+
+    def download_and_clean(self, comp_name, comp_type, project_dir):
+        """
+        Downloads component zip file file from GitHub and unarchives it in current directory.
+
+        Extracts the contents of the zip file into a temporary diretory and moves over them to the current directory.
+
+        Parameters
+        ----------
+          comp_name(string): Name of the component.
+          comp_type(string): Type of the component(template or repository).
+
+        Returns
+        -------
+            None
+        """
+        comp_url = self.get_download_url(comp_name, comp_type)
+
+        download_response = requests.get(comp_url, stream=True)
+        if download_response.status_code != 200:
+            try:
+                download_response.raise_for_status()
+            except Exception as e:
+                logging.error(e)
+                raise e
+            finally:
+                raise Exception(error_messages.INIT_FAILS_DURING_COMPONENT_DOWNLOAD.format(comp_type))
+
+        logging.debug("Downloading the component {}...".format(comp_type))
+        with zipfile.ZipFile(BytesIO(download_response.content)) as zfile:
+            with tempfile.TemporaryDirectory() as tmpdirname:
+                # Extracts the zip file into temporary directory - /some-temp-dir/downloaded-zip-folder/
+                zfile.extractall(tmpdirname)
+                # Moves the unarchived contents from temporary folder (downloaded-zip-folder) to current directory.
+                for f in Path(tmpdirname).joinpath(zfile.namelist()[0]).iterdir():
+                    shutil.move(str(f), project_dir)
```

Comment:
This could throw IO error

Reply:
We wrapped entire function `download_and_clean` under try-catch. Also, even before trying to download, we checked if the `project_dir` is empty or doesn't exist.